docs/src/pages/demos/text-fields/TextFields.js

```diff
@@ -246,6 +246,18 @@ class TextFields extends React.Component {
           defaultValue="Bare"
           margin="normal"
         />
+        <TextField
+          id="standard-name-with-character-limit"
+          label="Name"
+          className={classes.textField}
+          value={this.state.name}
+          onChange={this.handleChange('name')}
+          margin="dense"
+          characterLimit={15}
+          inputProps={{
+            maxlength: '15',
+          }}
```

Comment:
:+1: Validation should not be handled by the component.

Reply:
> I guess this should include the ability to have characterCount in a controlled state since we have to deal with https://blog.jonnew.com/posts/poo-dot-length-equals-two.

Even if it was controlled and we are checking it in code when `value` changes this would still exist `"💩".length === 2`

Why is `maxlength` approach not better, It is built in native feature of `input`

Reply:
> Why is maxlength approach not better, It is built in native feature of input
I didn't have any objections to this approach. This was the right call.

Concerning making this controlled: If it was controlled then the component should not override it with it's computed value. That's what meant by controlled.

Reply:
Oh understood! Would it be a better approach to just add this in the docs, one `characterLimit` example with the `maxlength` attribute for uncontrolled component, & the other one where the `TextField` component is controlled and we implement the `maxlength` functionality in the `onChange` method?

Reply:
@eps1lon sir I never got feedback on this. 😅

Reply:
@adeelibr I guess this is what I meant. Go ahead.

Reply:
Sir I tested this out with an uncontrolled component, but since the `Textfield` is uncontrolled an it doesn't have a value prop for it to have, the `characterLimit` functionality won't work. Because it need the `value.length` to find the remaining character limit.

I have updated my PR can you please have a quick glance on it.

Reply:
What I meant was that I have `characterLimit` and `characterCount` props. If no `characterCount` is supplied then fallback to `value.length.`